ktor-server/ktor-server-core/src/io/ktor/config/ApplicationConfig.kt

```diff
@@ -8,15 +8,15 @@ import io.ktor.util.*
 @KtorExperimentalAPI
 interface ApplicationConfig {
     /**
-     * Get config property with [path] or fail
+     * Get config property with [path], return [default] or fail if no [default] provided
      * @throws ApplicationConfigurationException
      */
-    fun property(path: String): ApplicationConfigValue
+    fun property(path: String, default: ApplicationConfigValue? = null): ApplicationConfigValue
 
     /**
-     * Get config property value for [path] or return `null`
+     * Get config property value for [path], return [default] or return `null`
      */
-    fun propertyOrNull(path: String): ApplicationConfigValue?
+    fun propertyOrNull(path: String, default: ApplicationConfigValue? = null): ApplicationConfigValue?
```

Comment:
Then it should be named `propertyOrDefault`. However I'd keep `propertyOrNull` as well with the default implementation like this: `fun propertyOrNull(path: String) = propertyOrDefault(path, null)`

Reply:
One of the ideas was to provide a method, which returns a null-safe object.
Please, consider this commit: https://github.com/ktorio/ktor/pull/735/commits/a505f85a75182f60133d0b404618e25f6423c682.

Reply:
The main question is remaining: why one couldn't simply always use elvis instead?

Reply:
The same reason Kotlin collections includes symmetric method, like `isEmpty()`/`isNotEmpty()`: convenience and ability to write a bit more clean code.
Using of `propertyOrNull()` with elvis operator to use some default looks like a tricky solution and complicates a code a bit.

However, if we could still consider the first commit: it would make more sense to have an existing methods `property()`/`propertyOrNull()`, which can be used "as is" or with a default value. There is a good reason to have separate methods with two different reactions: throw an expection, or return null/error code, but `propertyOrDefault()` seems too much. Defaults are convenient as optional parameters.